src/librustc_mir/interpret/step.rs

```diff
@@ -35,18 +35,40 @@ fn binop_right_homogeneous(op: mir::BinOp) -> bool {
     }
 }
 
+#[derive(Eq, PartialEq, Debug)]
+pub enum StepOutcome {
+    /// Indicates that there is still more work to do.
+    /// step() should be called again
+    MoreToDo,
+    /// Indicates that step() has completed all of
+    /// the available work - there is no point in making
+    /// further calls to step()
+    Done,
+    /// Indicates that a 'Resume' terminator
+    /// has been hit. This should only occur when
+    /// explicitly handling unwinding.
+    /// This requires special handling by the caller.
+    /// Callers not expecting it should return an error
+    /// to their caller, or panic otherwise.
+    Resume
+}
```

Comment:
This looks like you are still driving your own `run()` loop? I don't think this enum should be needed.

Reply:
No - there's still only one `run()` loop. This is just to give extra information to tools like `priroda`, or anything that implements its own `run()` loop. If you look at my changes to `run()`, you'll see that `Resume` is treated just like `MoreToDo`

Reply:
At the least then this should be `MoreToDo { resume: bool }`.

@oli-obk how would prioroda benefit from knowing if the last executed step was a `Resume` or not? I mean sure it has to visualize all sorts of things, but it seems oddly specific to me to single out this one distinction. Why not also single out if this was a terminator or a statement, or so?

Reply:
uh, I don't think we need any special info. All that priroda wants is that calling `step` will never do more than one step, everything else works just like miri.

Reply:
Okay, in that case @Aaron1011 please revert this to how it was (unless there is another good reason for this change of course).